src/features/comments.ts

```diff
@@ -0,0 +1,114 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchComments = createAsyncThunk<Comment[], Post['id']>(
+  'comments/fetch',
+  async (postId, { rejectWithValue }) => {
+    try {
+      return await getPostComments(postId);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to fetch comments',
+      );
+    }
+  },
+);
+
+export const addComment = createAsyncThunk<Comment, Omit<Comment, 'id'>>(
+  'comments/add',
+  async (comment, { rejectWithValue }) => {
+    try {
+      return await createComment(comment);
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to add comment',
+      );
+    }
+  },
+);
+
+export const removeComment = createAsyncThunk<number, Comment['id']>(
+  'comments/remove',
+  async (commentId, { rejectWithValue }) => {
+    try {
+      await deleteComment(commentId);
+
+      return commentId;
+    } catch (error) {
+      return rejectWithValue(
+        (error as Error).message || 'Failed to remove comment',
+      );
+    }
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(fetchComments.pending, state => ({
+        ...state,
+        loaded: false,
+        hasError: false,
+      }))
+      .addCase(fetchComments.fulfilled, (state, action) => ({
+        ...state,
+        comments: action.payload,
+        loaded: true,
+        hasError: false,
+      }))
```

Comment:
Similarly, when fetching comments is fulfilled, you are replacing the entire `comments` array with the new comments. If you have more than one post, this could lead to comments from other posts being lost. Consider appending new comments to the existing array instead of replacing it.